src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -153,6 +153,11 @@ public Result CalculateBasedlinedResult(DictionaryMergeBehavior propertyBagMerge
                 resultMatchingProperties.Add(MatchedResults.MatchResultMetadata_PreviousGuid, PreviousResult.Result.Guid);
             }
 
+            if (PreviousResult.Result.Suppressions != null && CurrentResult.Result.Suppressions == null)
+            {
+                result.Suppressions = PreviousResult.Result.Suppressions;
+            }
```

Comment:
this will flow suppressions from the baseline file to the current file.

Reply:
so here we don't consider current file's suppression?

Reply:
Since we start from the current, we just have to compare previous with current. if previous exist and current does not, we should replace.
otherwise, we will keep the current :)